src/DynamoCore/Models/AnnotationModel.cs

```diff
@@ -323,13 +355,37 @@ protected override void DeserializeCore(XmlElement element, SaveContext context)
                      if (SelectedModels != null)
                      {
                          var result = mhelper.ReadGuid("ModelGuid", new Guid());
-                        var model = SelectedModels.FirstOrDefault(x => x.GUID == result);
+                         ModelBase model = null;
+                         if(GetModelBaseEvent != null)
```

Comment:
This should be a separate function called `OnModelBaseRequested` and do the null check there, this is a useful convention that prevents duplicate code.